File: golangLeetcode/122.买卖股票的最佳时机-ii.go
/*
 * @lc app=leetcode.cn id=122 lang=golang
 *
 * [122] 买卖股票的最佳时机 II
 */

// @lc code=start
func maxProfit(prices []int) int {
	if len(prices) <= 1 {
		return 0
	}
	maxVal := 0
	lastMin := prices[0]
	for _, val := range prices {
		if val > lastMin {
			maxVal += (val - lastMin)
		}
		lastMin = val

	}
	return maxVal

}

// @lc code=end

File: LoveSwiftLeetcode/122.买卖股票的最佳时机-ii.swift
/*
 * @lc app=leetcode.cn id=122 lang=swift
 *
 * [122] 买卖股票的最佳时机 II
 */

// @lc code=start
class Solution {
    func maxProfit_0(_ prices: [Int]) -> Int {
        var curMin = 0, curMax = Int.max
        var profit = 0
        for i in prices {
            var needC = false
            if i > curMax {
                curMin = curMax
                curMax = i
                needC = true
            }else if i < curMax {
                curMin = i
                curMax = i
            }
            if needC {
                profit += (curMax - curMin)
            }
        }
        return profit
    }
}
// @lc code=end

